packages/extension/src/components/pages/BuildTransaction/BuildTransaction.tsx

```diff
@@ -0,0 +1,174 @@
+
+import {
+  SETTINGS_PATH,
+  SIGN_TRANSACTION_PATH,
+  STORAGE_MESSAGING_KEY,
+  SUBMIT_TRANSACTION_PATH
+} from '../../../constants';
+import { useNetwork, useWallet } from '../../../contexts';
+import { generateKey, saveInChromeSessionStorage } from '../../../utils';
+import { PageWithReturn } from '../../templates';
+
+export const BuildTransaction: FC = () => {
+  const { getCurrentWallet } = useWallet();
+  const { hasOfflineBanner } = useNetwork();
+  const editorHeight = useMemo(() => {
+    return hasOfflineBanner ? '250px' : '310px';
+  }, [hasOfflineBanner]);
+
+  const navigate = useNavigate();
+  const wallet = getCurrentWallet();
+
+  const [jsonInput, setJsonInput] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const handleBack = useCallback(() => {
+    navigate(SETTINGS_PATH);
+  }, [navigate]);
+
+  const handleSignTransaction = useCallback(() => {
+    const key = generateKey();
+    try {
+      JSON.parse(jsonInput);
+    } catch (e) {
+      setErrorMessage('Invalid JSON');
+      return;
+    }
+
+    try {
+      const parsed = JSON.parse(jsonInput);
+      if (!('Account' in parsed)) {
+        parsed.Account = wallet?.publicAddress;
+      }
+
+      validate(parsed);
+      saveInChromeSessionStorage(key, JSON.stringify({ transaction: parsed })).then(() =>
+        navigate(
+          `${SIGN_TRANSACTION_PATH}?${STORAGE_MESSAGING_KEY}=${key}&inAppCall=true&sign=transaction`
+        )
+      );
+    } catch (e) {
+      setErrorMessage('Invalid transaction');
+      return;
+    }
+    setErrorMessage('');
+  }, [jsonInput, navigate, wallet?.publicAddress]);
+
+  const handleSubmitTransaction = useCallback(() => {
+    const key = generateKey();
+    try {
+      JSON.parse(jsonInput);
+    } catch (e) {
+      setErrorMessage('Invalid JSON');
+      return;
+    }
+
+    try {
+      const parsed = JSON.parse(jsonInput);
+      if (!('Account' in parsed)) {
+        parsed.Account = wallet?.publicAddress;
+      }
+
+      validate(parsed);
+      saveInChromeSessionStorage(key, JSON.stringify({ transaction: parsed })).then(() =>
+        navigate(
+          `${SUBMIT_TRANSACTION_PATH}?${STORAGE_MESSAGING_KEY}=${key}&inAppCall=true&submit=transaction`
+        )
+      );
+    } catch (e) {
+      setErrorMessage('Invalid transaction');
+      return;
+    }
+    setErrorMessage('');
+  }, [jsonInput, navigate, wallet?.publicAddress]);
+
+  const handleBeautifyJson = useCallback(() => {
+    try {
+      const parsedJson = JSON.parse(jsonInput);
+      const beautifiedJson = JSON.stringify(parsedJson, null, 2);
+      setJsonInput(beautifiedJson);
+      setErrorMessage('');
+    } catch (e) {
+      setErrorMessage('Invalid JSON - cannot beautify');
+    }
+  }, [jsonInput]);
+
+  const jsonEditorStyle = `
+    .json-editor {
+      outline: none;
+    }`;
+
+  return (
+    <PageWithReturn title="Build Transaction" onBackClick={handleBack}>
```

Comment:
I would rename it sign transaction locally, build doesn't really mean anything.

Reply:
I have renamed into ''Submit Raw Transaction''